tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -606,6 +606,40 @@ static PyObject * THPVariable_float(PyObject* self, PyObject* args, PyObject* kw
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject * THPVariable_cdouble(PyObject* self, PyObject* args, PyObject* kwargs) {
```

Comment:
is there a reason why we need manual cpp bindings here? why can't we just add an entry to `native_functions.yaml` and make it a method only?

Reply:
I have no idea. I just mimicked the implementation of `.float()` and `.double()`.

Reply:
I think we'd prefer to make cfloat / cdouble into native functions (assuming PyTorch codegen works on them). This way, we don't have to worry about manually implementing support for other pytorch subsystems (like TorchScript).

Looking into the history of `Tensor.float()` and co. a little bit...
It looks like THPVariable_float and co came into existence in https://github.com/pytorch/pytorch/pull/3762 and haven't been modified heavily since. Back in the days of https://github.com/pytorch/pytorch/pull/3762 it looks like native_functions.yaml wasn't fully built out yet: https://github.com/gchanan/pytorch/blob/var_conv_functions/aten/src/ATen/native/native_functions.yaml.

@gchanan do you know if there are any blockers to making `Tensor.float(), Tensor.double(), ...` real native functions?